app/facades/food_facade.rb

```diff
@@ -0,0 +1,15 @@
+class FoodFacade
+
+  def self.ingredient_search(search_string)
+    data = FoodService.foods_by_ingredient(search_string)
+
+    food_results = data[:foods]
+
+    @foods = food_results.map do |food|
+      Food.new(food)
+    end
+
+    [@foods.first(10),data[:totalHits]]
```

Comment:
This is a great way to handle the case in which you need both of these objects! 

We wouldn't expect anyone to use this in an assessment but just to give you another option for this case: you could use [OpenStruct](https://ruby-doc.org/stdlib-1.9.3/libdoc/ostruct/rdoc/OpenStruct.html) to return both these values. This is used a lot in the industry for this type of case when you could use a PORO but don't necessary need to because you're only creating it as a way to hold these items, not as a domain object to store state like a Food PORO. Instead of creating a whole class, OpenStruct allows you to create a one-time hash-like object where you can store attributes with keys. The benefit of this over an array is that you can then reference, in your controller, `results.food_collection` and `results.total_hits` rather than using `results.first` and `.last`, which can be tricky in case you mess up what order those results come in. 

Again, using an array is a GREAT solution but this is just another option for the future!

Reply:
Oh very cool, I'll definitely check out OpenStruct!